.github/workflows/test.yml

```diff
@@ -13,7 +13,57 @@ jobs:
       - uses: denoland/setup-deno@v1
         with:
           deno-version: v1.x
-      - run: KAKAROT_ADDRESS=0x1 deno test --allow-env
+      # Checkout target branch and run tests
+      - name: Checkout target branch
+        run: |
+          git fetch origin ${{ github.head_ref }}
+          git checkout ${{ github.head_ref }}
+      - name: Run Tests with Coverage on target branch
+        run: KAKAROT_ADDRESS=0x1 deno test --allow-env --coverage=cov_profile
+      - name: Generate Coverage Report on target branch
+        run: deno coverage cov_profile --lcov --output=cov_profile.lcov
+      - name: Install lcov
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y lcov bc
+      - name: Generate HTML report from lcov for the target branch
+        run: |
+          genhtml -o cov_profile/html cov_profile.lcov
```

Comment:
I don't know if that would make everything easier but deno supports generating html for the coverage with `deno coverage --html`.

Reply:
fixed, thanks